posts/test_comment.py

```diff
@@ -0,0 +1,118 @@
+                        test test test test test test test test
+                        test test test test test test test test test test test test test test test
+                        test test test test test test test test test test test test test test test test
+                        test test test test test test test test test test test test test test test test
+                        test test test test test test test test test test test test test test test test test test
+                        test test test test test test test test test test test test test test test test test test test
+                        test test test test test test test test test test test test test test test test test test test
+                    """
+
+# -----------------------------------------------fixtures----------------------------------------------------------
+
+
+@pytest.fixture
+def post_creator():
+    return User(username=USERNAME, first_name=FIRSTNAME, last_name=LASTNAME, password=PASSWORD, email=EMAIL)
+
+
+@pytest.fixture
+def persist_post_creator(post_creator):
+    post_creator.save()
+    return post_creator
+
+
+@pytest.fixture
+def new_resume(post_creator):
+    return Resume(author=post_creator, description=DESCRIPTION)
+
+
+@pytest.fixture
+def persist_resume(new_resume):
+    new_resume.author.save()
+    new_resume.save()
+    return new_resume
+
+
+@pytest.fixture
+def comment_creator():
+    return User(username=COMMENT_USERNAME,
+                first_name=COMMENT_FIRSTNAME,
+                last_name=COMMENT_LASTNAME,
+                password=COMMENT_PASSWORD,
+                email=COMMENT_EMAIL)
+
+
+@pytest.fixture
+def new_comment(comment_creator, new_resume):
+    return Comment(post=new_resume, author=comment_creator, comment_text=COMMENT)
+
+
+@pytest.fixture
+def persist_comment(new_comment):
+    new_comment.post.author.save()
+    new_comment.post.save()
+    new_comment.author.save()
+    new_comment.save()
+    return new_comment
+
+
+@pytest.mark.django_db()
+class TestComment:
+
+    # Check if the Comment values are the same as comment input
+    def test_save_comment(self, new_comment):
+        assert new_comment.post.author.username == USERNAME
+        assert new_comment.author.username == COMMENT_USERNAME
+        assert new_comment.comment_text == COMMENT
+
+    # Check if the comment is saved in the database
+    def test_persist_comment(self, persist_comment):
+        assert persist_comment in Comment.objects.all()
+
+    # Check if the comment is deleted in the database
+    def test_delete_comment(self, persist_comment):
+        comment = persist_comment
+        comment.delete()
+        assert comment not in Comment.objects.all()
+
+    # Check if deletion of resume delete the comment from database
+    def test_comment_deletion_after_resume_deletion(self, persist_comment):
+        comment = persist_comment
+        comment.post.delete()
+        assert comment.post not in Resume.objects.all()
```

Comment:
I think we dont need line 105 because we check the deletion of resume in resume tests

Reply:
You are right that we checked the deletion of resume in resume test, I've wanted to check verify the resume is deleted before verifying the deletion of the comment.